tests/regression/models/solid_mechanics/regression_chaboche.cxx

```diff
@@ -26,101 +26,17 @@
-  auto mandel_stress = std::make_shared<IsotropicMandelStress>("mandel_stress");
-  auto isoharden = std::make_shared<VoceIsotropicHardening>("isotropic_hardening", R, d);
-
-  auto sm = std::make_shared<J2StressMeasure>("stress_measure");
-  auto yield =
-      std::make_shared<IsotropicAndKinematicHardeningYieldFunction>("yield_function", sm, s0);
-
-  std::vector<std::string> bs_name_1{"state", "internal_state", "backstress_1"};
-  std::vector<std::string> bs_name_2{"state", "internal_state", "backstress_2"};
-
-  auto bs1 = std::make_shared<ChabochePlasticHardening>("chaboche_1", C1, g1, A1, a1, "_1");
-  auto bs2 = std::make_shared<ChabochePlasticHardening>("chaboche_2", C2, g2, A2, a2, "_2");
-
-  std::vector<std::vector<std::string>> bs_names({bs_name_1, bs_name_2});
-  std::vector<std::string> kinname({"state", "hardening_interface", "kinematic_hardening"});
-
-  auto kinharden = std::make_shared<SumModel<SymR2>>("kinharden", bs_names, kinname);
-
-  auto direction =
-      std::make_shared<AssociativePlasticFlowDirection>("plastic_flow_direction", yield);
-  auto eeprate = std::make_shared<AssociativeIsotropicPlasticHardening>("eeprate", yield);
-  auto hrate = std::make_shared<PerzynaPlasticFlowRate>("hardening_rate", eta, n);
-  auto Eprate = std::make_shared<PlasticStrainRate>("plastic_strain_rate");
-
-  auto rate = std::make_shared<ComposedModel>("rate",
-                                              std::vector<std::shared_ptr<Model>>{Erate,
-                                                                                  Eerate,
-                                                                                  elasticity,
-                                                                                  mandel_stress,
-                                                                                  isoharden,
-                                                                                  bs1,
-                                                                                  bs2,
-                                                                                  kinharden,
-                                                                                  yield,
-                                                                                  direction,
-                                                                                  eeprate,
-                                                                                  hrate,
-                                                                                  Eprate});
-
-  auto implicit_rate = std::make_shared<ImplicitTimeIntegration>("implicit_time_integration", rate);
-  auto solver = std::make_shared<NewtonNonlinearSolver>(params);
-  auto model = std::make_shared<ImplicitUpdate>("viscoplasticity", implicit_rate, solver);
+  HITParser parser;
+  parser.parse_and_manufacture("regression/models/solid_mechanics/regression_chaboche.i");
```

Comment:
How about we have a helper function at least for the tests

`load_hit_model(std::string path)`

or 

`load_model(std::string path, ParserType type)`